### PR TITLE
Depend on transformers-compat

### LINKED ISSUE #38: Compatibility with older transformers
Using the `transformers-compat` package it's possible to support older `transformers` with no effort (no changes to code necessary, only the `.cabal` file needs to be modified).

### PR DESCRIPTION
**Pull request by int-index**

### CODE REVIEW

File: errors.cabal [hunk 1]
```diff
@@ -24,7 +24,8 @@ Library
     Build-Depends:
         base         >= 4     && < 5  ,
         safe         >= 0.3.3 && < 0.4,
-        transformers >= 0.4   && < 0.5
+        transformers >= 0.2   && < 0.5,
+        transformers-compat >= 0.4
```
Comment (Gabriel439) [inline errors.cabal]: Could you also add an upper bound on the `transformers-compat` dependency?

**Comment by int-index** [comment]
I've put the `< 1` upper bound because it's what `lens` does.

**Comment by Gabriel439** [comment]
`transformers-compat` is one package where I'm extra picky about keeping the upper bound low because it's been notorious for build failures in other libraries in the past due to lax upper bounds.  I'd prefer narrowing it to `< 0.5`.

**Comment by int-index** [comment]
Let me squash those commits so they don't clutter the history.

**Comment by Gabriel439** [comment]
Sounds good

**Comment by int-index** [comment]
Done

**Comment by Gabriel439** [comment]
Thank you!  I really appreciate all your work!

**Comment by Gabriel439** [comment]
I will go ahead and upload this if there aren't any further changes pending

**Comment by int-index** [comment]
Sure, go ahead. Thanks for the kind words :)

**Pull request merged**